src/main/java/org/jenkinsci/plugins/github_branch_source/WildcardPullRequestLabelFilterTrait.java

```diff
@@ -0,0 +1,168 @@
+                          "%n    Won't Build PR %s. PR has no labels and inclusion does not match '*'.%n",
+                          "#" + prHead.getNumber());
+                  return true;
+                }
+                return false;
+              }
+              // If any labels match the excludes the pr build will be excluded
+              if (!excludes.isEmpty()) {
+                Predicate<String> exclusionPredicate = getPattern(getExcludes()).asPredicate();
+                for (String prLabel : prLabels) {
+                  if (exclusionPredicate.test(prLabel)) {
+                    request
+                        .listener()
+                        .getLogger()
+                        .format(
+                            "%n    Won't Build PR %s. Label %s is marked for exclusion.%n",
+                            "#" + prHead.getNumber(), prLabel);
+                    return true;
+                  }
+                }
+              }
+              // If all labels are included the don't exclude the PR
+              if (allLabelsIncluded) {
+                return false;
+              }
+              // Otherwise if any of the labels match the includes the pr build will be included
+              if (prLabels.stream().anyMatch(getPattern(getIncludes()).asPredicate())) {
+                return false;
+              }
+              // no labels match the includes therefore the PR will be excluded
+              request
+                  .listener()
+                  .getLogger()
+                  .format(
+                      "%n    Won't Build PR %s. No labels match inclusion [%s].%n",
+                      "#" + prHead.getNumber(), prLabels);
+              return true;
+            }
+            return false;
+          }
+        });
+  }
+
+  /**
+   * Returns the pattern corresponding to the labels containing wildcards.
+   *
+   * @param labels the names of labels to create a pattern for
+   * @return pattern corresponding to the labels containing wildcards
+   */
+  private Pattern getPattern(String labels) {
+    StringBuilder quotedLabels = new StringBuilder();
+    for (String wildcard : labels.split(" ")) {
+      StringBuilder quotedLabel = new StringBuilder();
+      for (String label : wildcard.split("(?=[*])|(?<=[*])")) {
+        if (label.equals("*")) {
+          quotedLabel.append(".*");
+        } else if (!label.isEmpty()) {
+          quotedLabel.append(Pattern.quote(label));
+        }
+      }
+      if (quotedLabels.length() > 0) {
+        quotedLabels.append("|");
+      }
+      quotedLabels.append(quotedLabel);
+    }
+    return Pattern.compile(quotedLabels.toString());
+  }
+
+  @Symbol({"labelPullRequestFilter"})
+  @Extension
+  @Selection
+  public static class DescriptorImpl extends SCMSourceTraitDescriptor {
+    public DescriptorImpl() {}
+
+    public String getDisplayName() {
+      return Messages.WildcardPullRequestLabelFilterTrait_DisplayName();
+    }
+  }
```

Comment:
Please make this descriptor only applicable to GitHub branch sources, see [JENKINS-66503](https://issues.jenkins.io/browse/JENKINS-66503).

Reply:
This has been addressed and the ignore drafts trait has been fixed here https://github.com/jenkinsci/github-branch-source-plugin/pull/458